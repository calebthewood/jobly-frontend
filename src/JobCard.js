import { useContext, useState } from "react";
import UserContext from "./UserContext";
import JoblyApi from './api';

/** Creates card for each Job
 *
 * prop - {job}
 *
 * state - none
 *
 * JobList -> JobCard
 */
function JobCard({ job }) {
  const { currentUser } = useContext(UserContext);

  const active = () => {
    if (currentUser?.jobs) {
      return currentUser.jobs.includes(job.id)
    } else {
      return false;
    }
  }

  const [applied, setApplied] = useState(active);


  async function handleApply() {
    if (!applied) {
      await JoblyApi.applyToJob(job.id);
      setApplied(() => !applied);
    }
  }

  const applyBtn = <button
    style={{ width: "100px" }}
    type="button"
    className="btn btn-danger"
    onClick={handleApply}>Apply</button>;

  const unApplyBtn = <button
    style={{ width: "100px" }}
    type="button"
    className="btn btn-danger"
    onClick={handleApply}
    disabled>Applied</button>;

  return (
    <div className="row justify-content-center">
      <div className="col-12 col-md-10 col-xl-7">
        <div className="card shadow mb-4 col">
          <div className="card-body">
            <h6 className="card-title">{job.title}</h6>
            <p className="card-text fw-light">{job.companyName}</p>

            <div className="row justify-content-between card-text fw-light">
              <div className="col col-9">
                <small>
                  Salary: {job.salary || ""}
                </small>

                <div className="card-text fw-light">
                  <small>
                    Equity: {job.equity || ""}
                  </small>
                </div>
              </div>

              <div className="col-3">
                { applied ? unApplyBtn : applyBtn}
              </div>
            </div>

          </div>
        </div>
      </div>
    </div>

  );
}

export default JobCard;