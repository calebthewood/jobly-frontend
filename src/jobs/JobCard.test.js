import { render } from "@testing-library/react";
import JobCard from "./JobCard";
import { UserProvider } from "../testUtils";


describe('CompanyDetail', () => {
  it("matches snapshot", function () {
    let item = { title: "CEO", salary: 1000000, equity: 10 };
    const { asFragment } = render(
      <UserProvider>
        <JobCard job={item} />
      </UserProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});