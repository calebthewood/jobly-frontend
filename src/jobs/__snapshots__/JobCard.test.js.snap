// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-12 col-md-10 col-xl-7"
    >
      <div
        class="card shadow mb-4 col"
      >
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            CEO
          </h6>
          <p
            class="card-text fw-light"
          />
          <div
            class="row card-text fw-light"
          >
            <div
              class="col"
            >
              <small>
                Salary: $1,000,000
              </small>
              <div
                class="card-text fw-light"
              >
                <small>
                  Equity: 10
                </small>
              </div>
            </div>
            <div
              class="col align-self-end"
            >
              <button
                class="btn btn-danger float-end"
                style="width: 90px;"
                type="button"
              >
                Apply
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
